addon/lib/app.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * This Source Code is subject to the terms of the Mozilla Public License
+ * version 2.0 (the 'License'). You can obtain a copy of the License at
+ * http://mozilla.org/MPL/2.0/.
+ */
+
+const { Class } = require('sdk/core/heritage');
+const { emit, setListeners } = require('sdk/event/core');
+const { EventTarget } = require('sdk/event/target');
+const { PageMod } = require('sdk/page-mod');
+const { Request } = require('sdk/request');
+const { setInterval, clearInterval } = require('sdk/timers');
+
+const App = Class({ // eslint-disable-line new-cap
+  implements: [EventTarget],
+  initialize: function(options) {
+    setListeners(this, options);
+    this.baseUrl = options.baseUrl;
+    this.badgeColor = options.badgeColor;
+    this.experiments = {};
+    this.workers = new Set();
+    this.reloadInterval = setInterval(
+      () => this.loadExperimentList(),
+      options.reloadInterval
+    );
+
+    const pageIncludes = this.baseUrl + '/*';
+    this.page = new PageMod({
+      include: pageIncludes,
+      contentScriptFile: './message-bridge.js',
+      contentScriptWhen: 'start',
+      attachTo: ['top', 'existing']
+    });
+    this.page.on(
+      'attach',
+      worker => {
+        this.workers.add(worker);
+        worker.on('detach', () => this.workers.delete(worker));
+        worker.port.on('from-web-to-addon', ev => emit(this, ev.type, ev.data));
+      }
+    );
+    const beaconIncludes = (pageIncludes + ',' + options.whitelist).split(',');
+    this.beacon = new PageMod({
+      include: beaconIncludes,
+      contentScriptFile: './set-installed-flag.js',
+      contentScriptWhen: 'start',
+      attachTo: ['top', 'existing'],
+      contentScriptOptions: {
+        version: options.addonVersion
+      }
+    });
+    this.loadExperimentList();
+  },
+  loadExperimentList: function() {
+    const r = new Request({
```

Comment:
You might try seeing what happens when you enable / disable an experiment. We might need the CSRF stuff to enable the [`DELETE` / `PUT` requests we send to the API for counting experiment installs](https://github.com/mozilla/testpilot/blob/master/addon/index.js#L299). 

But, I can't remember if we need CSRF for those. This might be a throwback to when we had FxA and needed to POST to an `/auth/logout` URL to log the user out.